lib/practice_exercises.rb

```diff
@@ -1,13 +1,65 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
```

Comment:
This works but since you're making new strings and arrays with:

```ruby
  a = string.split /[^A-Za-z]+/
  b = a.map {|x| x.split //}
  c = b.flatten 
```

You have O(n) space complexity.  Also please don't make all your variable names so generic.